File: scenarios/calibration/python-sales-by-genre/solution/solve.py
#!/usr/bin/env python3
from pathlib import Path

path = Path("/workspace/repository/sales_by_genre.py")
source = path.read_text(encoding="utf-8")
before = "JOIN InvoiceLine AS il ON il.TrackId = g.GenreId"
after = "JOIN InvoiceLine AS il ON il.TrackId = t.TrackId"
if source.count(before) != 1:
    raise SystemExit("seeded join was not found exactly once")
path.write_text(source.replace(before, after), encoding="utf-8")

File: scenarios/calibration/angular-reading-time/solution/solve.mjs
#!/usr/bin/env node
import fs from 'node:fs';

const path = '/workspace/repository/src/app/features/article/components/article-preview.component.ts';
const source = fs.readFileSync(path, 'utf8');
const needle = '  toggleFavorite(favorited: boolean): void {';
if (source.split(needle).length !== 2) {
  throw new Error('component insertion point was not found exactly once');
}
const method = `  getReadingTime(body: string): number {
    const words = body.trim().split(/\\s+/).filter(Boolean);
    return Math.max(1, Math.ceil(words.length / 200));
  }

`;
fs.writeFileSync(path, source.replace(needle, method + needle));
